internal/engine/uiresource/subscriber.go

```diff
@@ -62,13 +72,38 @@ func (s *Subscriber) OnChange(ctx context.Context, st store.RStore, summary stor
 			// If there's a current version but no last version of this resource,
 			// create it.
 			st.Dispatch(NewUIResourceCreateAction(current))
+			err := s.client.Create(ctx, current.DeepCopy())
```

Comment:
I think for `Create` and `Update`, you'll want to _not_ `DeepCopy()` and only dispatch after the API operation. It'll mutate the object (e.g. populate UID, creation timestamp, resource version on create; increment resource version on update) so that's the version we should pass to engine (and store here to ensure that optimistic concurrency on update works as expected)